subsys/net/ip/ipv6_nbr.c

```diff
@@ -2404,9 +2459,23 @@ static enum net_verdict handle_ra_input(struct net_pkt *pkt,
 			break;
 #endif
 		case NET_ICMPV6_ND_OPT_ROUTE:
-			NET_DBG("Route option skipped");
-			goto skip;
+			if (!IS_ENABLED(CONFIG_NET_ROUTE)) {
+				NET_DBG("Route option skipped");
+				goto skip;
+			}
+
+			/* RFC 4191, ch. 2.3 */
+			if (nd_opt_hdr->len == 0U || nd_opt_hdr->len > 3U) {
+				NET_ERR("DROP: Invalid %s length (%d)",
+					"route info opt", nd_opt_hdr->len);
+				goto drop;
+			}
 
+			if (!handle_ra_route_info(pkt, nd_opt_hdr->len)) {
+				goto drop;
+			}
```

Comment:
Should we place this to the else branch (so that it is available if CONFIG_NET_ROUTE is enabled)?

Reply:
I'm not sure if I follow, if CONFIG_NET_ROUTE is enabled the following condition should be compiled out (as it'll always evaluate to false):
```
if (!IS_ENABLED(CONFIG_NET_ROUTE)) {
	NET_DBG("Route option skipped");
	goto skip;
}
```
So in this scenario, we go directly to the Route info option processing (after the length check). I don't think we need an `else` condition here, unless it's a prefered way due to some coding standard rules.

But I might be missing something, could you elaborate, maybe I did not catch what were you referring to?

Reply:
Hmm, indeed, I am not sure what I was thinking here, never mind my comment.